src/main/resources/META-INF/resources/primefaces/overlaypanel/overlaypanel.js

```diff
@@ -105,9 +103,7 @@ PrimeFaces.widget.OverlayPanel = PrimeFaces.widget.DynamicOverlayWidget.extend({
         this._super();
 
         // see #setupDialogSupport
-        if (!this.cfg.appendTo) {
-            PrimeFaces.utils.removeDynamicOverlay(this, this.jq, this.id, $(document.body));
-        }
+        PrimeFaces.utils.removeDynamicOverlay(this, this.jq, this.id, $(document.body));
```

Comment:
the last param is the "appendTo", which may be configured to e.g. someId.
I think we must use PrimeFaces.utils.resolveDynamicOverlayContainer(this) like in datepicker?

Reply:
Also unsure if we need this as it extends DynamicOverlayWidget, which uses registerDynamicOverlay and auto-registers removeDynamicOverlay?

Reply:
OK I added the methods for `resolve` and `refresh` now calls `cleanupDyanmicOverlay` instead of destroy just like DatePicker.  I think we should extend `DynamicOverlay` because that's exactly what this is right?

Reply:
Good point Do we even need this since it does extend DynamicOverlay.  Let me try and pull those calls out and re-run the integration tests.

Reply:
OK you are right I removed those calls in `refresh` and `destroy` and all integration tests passing.  So this is all handled in DynamicOverlayWidget we must have just never refactored this guy.